src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/OrderStateMessage/index.spec.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { PropsWithChildren } from 'react';
+import { render, screen } from '@testing-library/react';
+import { CertificateFactory, ContractFactory, OrderFactory } from 'utils/test/factories/joanie';
+import { OrderState } from 'types/Joanie';
+import { IntlProvider, createIntl } from 'react-intl';
+import OrderStateMessage, { messages } from '.';
+
+const intl = createIntl({ locale: 'en' });
+
+/*
+  Order lifecycle:
+    * Draft: order has been created
+    * Signature needed: order with contract but without signatures.
+    * Submited: order information have been validated
+    * Pending: payment has failed but can be retried
+    * Validated:
+        * Waiting countersignature: order with contract missing the second signature.
+        * Completed (Validated with generated certificate)
+        * On going (Validated without generated certificate)
+    * Canceled: has been canceled
+*/
+describe('<DashboardItemOrder/>', () => {
+  const Wrapper = ({ children }: PropsWithChildren) => (
+    <IntlProvider locale="en">{children}</IntlProvider>
+  );
+
+  it.each([
+    OrderState.DRAFT,
+    OrderState.SUBMITTED, // TODO: check if we prefer a message like "waiting for payment"
+    OrderState.PENDING, // TODO: check if we prefer a message like "waiting for payment"
```

Comment:
what do you think @sampaccoud ?

Reply:
I would say it's not necessarily payment, there could be other conditions like validation of the contract... So maybe "pending review" or "pending approval"?

Reply:
About contract i added here two new status:
* draft with a contract but without signature: need signature (from the learner).
* validated with a contract but with only one signature: waiting for organization to countersign.